reivew of README content

bestbug456 opened the pull request:
Hello!

I was looking for something for using ESP Home with zigbee, I found out this repository which was kinda easy to include and build with it, but it was kinda of a exploration of the readme and the example. I'd like to submit a pull request to enhance the readability of the README while preserving all the original information (to the best of my understanding).   

Before proceeding, I have a few questions to ensure the changes align with the project's goals:   

* CSV File Requirement: The example mentions a CSV file, but it's unclear whether it's a required file or an optional optimization. Could you clarify if this file is necessary for the build process or if it's just an optional step for specific use cases?  
* Limitations I’m not entirely certain if all the listed limitations are still applicable. Could you confirm if the current list is up-to-date or if any details need updating?
     

Thank you for your excellent work on this project! I was also curious—did you consider submitting a feature proposal to the original ESP Home project for Zigbee integration?
Cheers

Commit:
reivew of README content

luar123 commented:
Thanks, readme could always be improved.
I will probably need some time for a detailed review, but to answer your questions:

> The example mentions a CSV file, but it's unclear whether it's a required file or an optional optimization. Could you clarify if this file is necessary for the build process or if it's just an optional step for specific use cases?

The partitions.csv is needed. You will get a validation error if it is not correct.

> Limitations I’m not entirely certain if all the listed limitations are still applicable. Could you confirm if the current list is up-to-date or if any details need updating?

I think it is up-to-date.

Please remove the batches on top, two are useless anyway and the 3rd is not really helping.

Commit:
docs: Update README with required partition table configuration

Removed GitHub badges from README and added important documentation about the mandatory partitions_zb.csv file that must be included in ESP32 configurations when using this component. The new section explains that this file is required for proper Zigbee operations and can be found in the component directory, defining the memory partitions needed for Zigbee functionality.

luar123 commented on README.md:
```diff
@@ -8,65 +12,74 @@
-* Coordinator devices
-* Binding config in yaml
-* Reporting config in yaml
-* Control device support like switches ([workaround](https://github.com/luar123/zigbee_esphome/discussions/18#discussioncomment-11875376))
```
Please keep this information.

luar123 commented on README.md:
```diff
@@ -80,23 +93,32 @@ zigbee:
+## Basic Mode
 
-### Basic mode
-By adding `components: all` the endpoint definition is generated automatically. Currently sensor, binary_sensor, light and switch components are supported. Because this is an external component the whole implementation is a bit hacky and likely to fail with some setups. Also it is not possible to tweak the generated definitions. Each entity creates a new endpoint. Lights are using the light clust ...
```
Please also keep the second half of this paragraph.

bestbug456 replied:
Sorry, I was sure I did the changes requested. I've updated all. Let me know if is ok I will proceed to resolve the merge conflict

luar123 reviewed (changes requested):
Overall looks nice, just some information got lost. Can you include the missing pieces?

Commit:
update doc as requested by the mantainer

Commit:
Merge PR #73 changes (README improvements from #85) and remove partition table note

- Reorganized README with table of contents
- Formatted features, limitations, and configuration sections
- Removed partition table requirement (no longer needed)
- Kept all recent changes from #85

bestbug456 committed:
Merge branch 'master' into pr73

bestbug456 committed:
Merge pull request #1 from bestbug456/pr73

Pr73
